source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -22,12 +23,21 @@ RouteEntryImplBase::RouteEntryImplBase(
     config_headers_.push_back(header_map);
   }
 
+  if (route.route().has_metadata_match()) {
+    const auto filter_it = route.route().metadata_match().filter_metadata().find(
+        Envoy::Config::MetadataFilters::get().ENVOY_LB);
+    if (filter_it != route.route().metadata_match().filter_metadata().end()) {
+      metadata_match_criteria_.reset(
+          new Envoy::Router::MetadataMatchCriteriaImpl(filter_it->second));
+    }
+  }
+
```

Comment:
would https://github.com/envoyproxy/envoy/commit/62441f9fe2933d4846cfc3d166e52a9178be755f work with thrift without further work?

Reply:
(this relates to @brian-pane's perf comment)

Reply:
I believe it will, yes.